budgetApp1/budgetApp1/AuthenticationRepository.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  AuthenticationRepository.swift
+//  budgetApp1
+//
+//  Created by Lehlohonolo Nkadimeng on 2024/04/30.
+//
+
+import UIKit
+
+protocol AuthenticationRepositoryType: AnyObject {
+    func createUser(fullName: String, password: String, emailAddress: String, username: String)
+    func loginUser(username: String, password: String) -> Bool
+    func fetchAllUsers() -> Result<[UserEntity], Error>
+}
+
+class AuthenticationRepository: AuthenticationRepositoryType {
+    
+    // MARK: - Variables
+    
+    private var userSignedUp = false
+    private let coreDataHandler = CoreDataHandler()
+    
+    // MARK: - Functions
+    
+    func fetchAllUsers() -> Result<[UserEntity], Error> {
+        coreDataHandler.fetchAllUsers()
+    }
+    
+    func createUser(fullName: String, password: String, emailAddress: String, username: String) {
+        coreDataHandler.createUser(fullName: fullName, password: password, emailAddress: emailAddress, username: username)
+    }
+    
+    func loginUser(username: String, password: String) -> Bool {
+        var successfulLogin = false
+        
```

Comment:
func loginUser(username: String, password: String) -> Bool {
MarcusNkadimeng marked this conversation as resolved.

        if !coreDataHandler.checkIfUserHasAccount(username: username, password: password) {
            print("Error fetching username: \(AuthError.failedToFetchUser)")
            return false
        } else {
            return true
        }
    }
    
    Does this not work? Just neater if we can get rid of that variable

Reply:
Valid point. Removing it.